<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1080</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>1041</width>
      <height>691</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="GeometricTransformation">
     <attribute name="title">
      <string>GeometricTransformation</string>
     </attribute>
     <widget class="QPushButton" name="uploadfile">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>upload file</string>
      </property>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>260</y>
        <width>1011</width>
        <height>381</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>梯形</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="trapezoidalGeometricTransformationImage"/>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_8">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>波浪</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="wavyGeometricTransformationImage"/>
         </item>
        </layout>
       </item>
       <item row="1" column="2">
        <layout class="QVBoxLayout" name="verticalLayout_9">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>圓形</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="circularGeometricTransformationImage"/>
         </item>
        </layout>
       </item>
       <item row="0" column="2">
        <layout class="QGridLayout" name="gridLayout_7">
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="radiusSpinBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.800000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_24">
           <property name="text">
            <string>半徑 (比例)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="0">
          <widget class="QLabel" name="label_21">
           <property name="text">
            <string>震幅 (pixel)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_22">
           <property name="text">
            <string>頻率 (rad/pixel)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_23">
           <property name="text">
            <string>相位 (rad) (0~6.28(2pi))</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="amplitudeSpinBox">
           <property name="value">
            <double>10.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="frequencySpinBox">
           <property name="value">
            <double>10.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="phaseSpinBox">
           <property name="maximum">
            <double>6.280000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.314000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="0" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>上底 (比例)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>下底 (比例)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>高 (比例)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="topBaseSpinBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.800000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="bottomBaseSpinBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.700000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="heightSpinBox">
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.900000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="">
      <property name="geometry">
       <rect>
        <x>690</x>
        <y>10</y>
        <width>331</width>
        <height>231</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Original</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGraphicsView" name="originalImage"/>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="DWTfusion">
     <attribute name="title">
      <string>DWTfusion</string>
     </attribute>
     <widget class="QPushButton" name="uploadfile_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>191</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>upload file</string>
      </property>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>1021</width>
        <height>541</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>image1</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="originalImage_2"/>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>融合結果</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="imageFusionImage"/>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_10">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>image2</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="originalImage_3"/>
         </item>
        </layout>
       </item>
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_8">
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="levelSpinBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.800000000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_25">
           <property name="text">
            <string>level </string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="SLICseg">
     <attribute name="title">
      <string>SLICseg</string>
     </attribute>
     <widget class="QPushButton" name="uploadfile_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>upload file</string>
      </property>
     </widget>
     <widget class="QWidget" name="gridLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>1021</width>
        <height>541</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="2">
        <layout class="QVBoxLayout" name="verticalLayout_14">
         <item>
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>SegResult</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="SLICImage"/>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_12"/>
       </item>
       <item row="1" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Original imamge</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QGraphicsView" name="originalimage_4"/>
         </item>
        </layout>
       </item>
       <item row="0" column="2">
        <layout class="QGridLayout" name="gridLayout_9">
         <item row="0" column="1">
          <widget class="QSpinBox" name="KspinBox">
           <property name="minimum">
            <number>100</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="singleStep">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="MspinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>40</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_27">
           <property name="text">
            <string>m</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="maxIterSpinBox">
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_28">
           <property name="text">
            <string>max_iter</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_26">
           <property name="text">
            <string>K</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1080</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
